### PR TITLE
refactor(catalog): update ods components lib version to 17.1.0

### PR DESCRIPTION
**Pull request by vovh**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #13000
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- ~~[ ] Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

## Description

Update ODS Components package to 17.1.0 in catalog app

### CODE REVIEW

File: packages/manager/apps/catalog/src/components/Error/ErrorButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import { OsdsButton } from '@ovhcloud/ods-components/react';
+import { ODS_BUTTON_VARIANT } from '@ovhcloud/ods-components';
+
+export const ErrorButtons = () => {
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager/apps/catalog/src/components/Error/ErrorButtons.tsx:8]: Where do you use this component inside catalog ?

**Review by aboungnaseng-ovhcloud — DISMISSED** [review]

File: packages/manager/apps/catalog/src/components/Error/ErrorButtons.tsx (hunk 1 above)
Comment (vovh) [reply packages/manager/apps/catalog/src/components/Error/ErrorButtons.tsx:8]: It got included back from rebase.
Gonna check.

**Review by aboungnaseng-ovhcloud — DISMISSED** [review]

File: packages/manager/apps/catalog/e2e/FilterAndSearch.e2e.ts [hunk 2]
```diff
@@ -70,46 +60,67 @@ const validateProductCategories = async (
 test('should filter results based on Bare Metal Cloud universe', async ({
   page,
 }) => {
-  await clickButtonByName(page, 'filterButton');
-  await clickCheckboxByName(page, 'checkbox-universe-BareMetalCloud');
-  await clickButtonByName(page, 'applyFilterButton');
+  await clickButtonByText(
+    page,
+    searchTranslation.manager_catalog_search_filter_button,
+  );
```
Comment (chipp972) [inline packages/manager/apps/catalog/e2e/FilterAndSearch.e2e.ts:66]:
```suggestion
  await page.getByText(searchTranslation.manager_catalog_search_filter_button).click();
```

No need to create a function for this

File: packages/manager/apps/catalog/e2e/FilterAndSearch.e2e.ts [hunk 3]
```diff
@@ -70,46 +60,67 @@ const validateProductCategories = async (
...
 
   const links = await page.$$('osds-link');
   const resetFilterLink = links[0];
   if (resetFilterLink) await resetFilterLink.click();
 
-  await waitForProducts(page, 40);
+  await waitForProducts(page, 41);
```
Comment (chipp972) [inline packages/manager/apps/catalog/e2e/FilterAndSearch.e2e.ts:107]: I think we should mock the call if it's not done : The goal is not to test the catalog API

File: packages/manager/apps/catalog/src/components/Filters/Filters.tsx [hunk 4]
```diff
@@ -172,7 +170,7 @@ const Filters: React.FC<FiltersProps> = ({
             {t('manager_catalog_filters_reset')}
           </OsdsLink>
           <OsdsButton
-            name="applyFilterButton"
+            id="applyFilterButton"
```
Comment (chipp972) [inline packages/manager/apps/catalog/src/components/Filters/Filters.tsx:173]: Do we really need an id on this ?

File: packages/manager/apps/catalog/src/components/SearchBar/SearchBar.tsx [hunk 5]
```diff
@@ -84,12 +91,12 @@ const SearchBar: React.FC<SearchbarProps> = ({
         </span>
         <span>
           <OsdsButton
-            name="filterButton"
+            id="filterButton"
```
Comment (chipp972) [inline packages/manager/apps/catalog/src/components/SearchBar/SearchBar.tsx:94]: Do we need an id on this button ?

File: packages/manager/apps/catalog/e2e/FilterAndSearch.e2e.ts (hunk 3 above)
Comment (vovh) [reply packages/manager/apps/catalog/e2e/FilterAndSearch.e2e.ts:107]: @aboungnaseng-ovhcloud If you can create a ticket about mocking data test for catalog pretty please.
i agree with @chipp972 on this point but just gonna scope it on another ticket.

**Review by aboungnaseng-ovhcloud — DISMISSED** [review]

**Review by chipp972 — DISMISSED** [review]

**Review by JacquesLarique — APPROVED** [review]

**Review by chipp972 — APPROVED** [review]

**Pull request merged**